plugins/rest/rest_test.go

```diff
@@ -255,48 +269,63 @@ func TestClientCert(t *testing.T) {
 	}
 	ts.start()
 	defer ts.stop()
-	tmpPem, err := ioutil.TempFile("", "client.pem")
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer os.Remove(tmpPem.Name())
-	tmpKey, err := ioutil.TempFile("", "client.key")
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer os.Remove(tmpKey.Name())
-	if _, err := tmpPem.Write([]byte(ts.clientCertPem)); err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	if err := tmpPem.Close(); err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	if _, err := tmpKey.Write([]byte(ts.clientCertKey)); err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	if err := tmpKey.Close(); err != nil {
-		t.Fatalf("Unexpected error: %v", err)
+
+	files := map[string]string{
+		"client.pem": string(ts.clientCertPem),
+		"client.key": string(ts.clientCertKey),
 	}
+
+	test.WithTempFS(files, func(path string) {
+		certPath := filepath.Join(path, "client.pem")
+		keyPath := filepath.Join(path, "client.key")
+
+		client := newTestClient(t, &ts, certPath, keyPath)
+
+		ctx := context.Background()
+		if _, err := client.Do(ctx, "GET", "test"); err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		// Scramble the keys in the server
+		ts.stop()
+		ts.start()
+
+		// Ensure the keys don't work anymore, make a new client as the url will have changed
+		client = newTestClient(t, &ts, certPath, keyPath)
+		if _, err := client.Do(ctx, "GET", "test"); err == nil {
```

Comment:
Nit: should we have a more specific assertion here to guarantee we're seeing a certificate error?

Reply:
Good catch, I meant to do that here 👍